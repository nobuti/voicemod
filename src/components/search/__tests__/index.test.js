import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import Search from "../";

describe("Seacrh", () => {
  const onChange = jest.fn();

  test("renders properly", () => {
    const { container } = render(<Search onChange={onChange} name="wadus" />);

    expect(container).toMatchSnapshot();
    expect(screen.queryByTestId("search")).toBeInTheDocument();
  });

  test("search icon shows properly", () => {
    render(<Search onChange={onChange} />);

    expect(screen.queryByTestId("search-icon")).toBeDefined();
  });

  test("placeholder shows properly", () => {
    const placeholder = "Search something";
    render(<Search placeholder={placeholder} onChange={onChange} />);

    expect(screen.queryByTestId("search-field").placeholder).toEqual(
      placeholder
    );
  });

  test("manages focus properly", () => {
    render(<Search onChange={onChange} />);

    fireEvent.focus(screen.getByTestId("search-field"));

    expect(screen.queryByTestId("search")).toHaveClass("is-focused");
  });

  test("reset button is shown properly", () => {
    render(<Search onChange={onChange} />);

    expect(screen.queryByTestId("search-reset")).toBeNull();

    fireEvent.change(screen.getByTestId("search-field"), {
      target: { value: "wadus" },
    });

    expect(screen.getByTestId("search-field").value).toEqual("wadus");
    expect(screen.queryByTestId("search-reset")).toBeDefined();
  });

  test("reset button resets the field value properly", () => {
    render(<Search onChange={onChange} />);

    fireEvent.change(screen.getByTestId("search-field"), {
      target: { value: "wadus" },
    });

    expect(screen.getByTestId("search-field").value).toEqual("wadus");
    expect(onChange).toHaveBeenCalledWith("wadus");

    fireEvent.click(screen.getByTestId("search-reset"));
    expect(onChange).toHaveBeenCalledWith("");
  });
});
