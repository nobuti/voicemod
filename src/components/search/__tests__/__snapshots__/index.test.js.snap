// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Seacrh renders properly 1`] = `
<div>
  <div
    class="Search"
    data-testid="search"
  >
    <i
      class="Search-icon"
      data-testid="search-icon"
    />
    <input
      class="Search-field"
      data-testid="search-field"
      name="wadus"
      placeholder=""
      type="text"
      value=""
    />
  </div>
</div>
`;
